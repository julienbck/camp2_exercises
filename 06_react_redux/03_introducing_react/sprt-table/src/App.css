.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}


table {
  color:#FFF;
  background-color:#FFF;
  width: 80%;
  box-shadow: -5px 5px 20px #333;
  margin: 0 auto;
  font-size: 12pt;
}
table th {
  background-color: #0080FF;
  height: 50px;
  padding: 0;
  margin: 0;
  text-transform: uppercase;
}
table th button {
  border-style: none;
  background: none;
  text-transform: uppercase;
  color:#FFF;
  font-size: 12pt;

}
table th button:focus {
  outline: none;

}
table th button:hover {
  background-color: #FFF;
  color:#0080FF;
  width: 100%;
  height: 100%;
  transition: background-color 0.2s, color 0.1s;
}
table a {
    text-decoration: none;
    color: #FFF;
}
table tr {
  background-color: #FFF;
}
table > tbody > tr:hover {
  background-color: #045FB4;
  color:#FFF;
  transition: background-color 0.3s, color 0.3s;
}
table > tbody > tr {
    color: #045FB4;
}
table td {
  padding: 10px;
}
.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
