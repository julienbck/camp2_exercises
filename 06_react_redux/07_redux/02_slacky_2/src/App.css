html, body, #root {
  height: 100%;
}

.App {
  text-align: center;
  display:flex;
  flex-direction: column;
  justify-content: flex-start;
  height: 100%;
  align-items: stretch;
  align-content: stretch;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #1e3b5F;
  height: auto;
  width: 100%;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}


.Login {
  padding: 20px;
}

.Login div {
  padding: 20px;
}

.Chat {
  flex: 2 1;
  display:flex;
  flex-flow: column;
  justify-content: space-between;
  padding-top: 5px;
  align-items: stretch;
  align-content: stretch;
  max-height: calc(100vh - 50px)
}

.Chat-messages {
  flex: 1 1 auto;
  overflow: auto;
  max-height: calc(100% - 50px)
}

.Chat-form {
  height: 50px;
  background: #ccc;
  display: flex;
}
.Chat-form form {
  display: flex;
  width: 100%;
}

.Chat-form input {
  flex: 1 0;
  outline: none;
}

.Chat-form button {
  height: 100%;
  margin: 0;
  padding: 20px;
  background-color: #007bff;
  color: white;
  border: none;
  -webkit-appearance: none;
  outline: none;
}
.Chat-form button:hover {
  background-color: #FFF;
  color:#007bff;
  transition: background-color 0.2s, color 0.2s;
}

.message-container {
  margin-bottom: 10px;
  width: 100%;
  text-align: left;
}
.subcontainer{
  color: #FFF;
  height: 2vw;
  vertical-align: middle;
  line-height: 2vw;
  background-color: #1e3b5F;
}
.subcontainer2{
  color: #1e3b5F;
  height: auto;
  padding: 10px;
  vertical-align: middle;
  line-height: 2vw;
  background-color: #FCFCFC;
}

.author {
  font-weight: bold;
  padding: 5px 10px;
}
.date {
  display: flex;
  font-size: 14px;
  font-style: italic;
  padding: 5px 10px;
}
