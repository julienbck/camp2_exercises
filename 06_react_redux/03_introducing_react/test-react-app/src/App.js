import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';

class App extends Component {
  render() {
    return (
          <h1 className="App-title">Hello {this.props.test} ! </h1>
    );
  }
}

export default App;
