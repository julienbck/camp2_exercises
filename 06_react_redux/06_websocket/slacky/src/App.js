import React, { Component } from 'react';
import './App.css';
import Chat from './Chat.js';


class App extends Component {
  constructor(props){
    super(props);
    this.state = {
      username: '',
      message: '',
      messages: [],
      user: []
    };
    this.updateUsername = this.updateUsername.bind(this);
    this.updateMessage = this.updateMessage.bind(this);
    this.updateBoardUsername = this.updateBoardUsername.bind(this);
  }
  componentDidMount() {
    // We open a connection from the client, related to the server
    const ws = new WebSocket("ws://localhost:4000");
    const nbUsersElem = document.getElementById("nbUsers");
    // What to do when we receive a message?
    ws.onmessage = (event) => {
      console.log("Message: " + event.data);
      nbUsersElem.innerHTML = event.data;
    };

    // Alert the server that the client is gone
    window.addEventListener("beforeunload", () => ws.send("CLOSE"));
  }
  updateUsername(event){
    this.setState({
      username: event.target.value
    });
  }
  updateMessage(message){
    this.setState({
      message: message.target.value,
      messages: this.state.messages + [message.target.value]
    });
  }
  updateBoardUsername (event){
    let push = this.state.users.push(event.target.value);
    this.setState({
      users: push
    })
  }
  render() {
    return(
      <div id="App">
      <p>There is <span id="nbUsers">0</span> users connected.</p>
        <form>
        <label>
        Username:
        <input type="text" onChange={this.updateUsername} value={this.state.username}></input>
        </label>
        <input type="submit" onClick={this.updateBoardUsername} value="Submit" />
        </form>
        <form>
        <label>
        Message:
        <input type="text" onClik={this.updateMessage} value={this.state.message}></input>
        </label>
        </form>
        <Chat username={this.state.username} message={this.state.message} />
        <h3>{this.state.messages}</h3>

        <h4>{this.state.users}</h4>
      </div>
    );
  }
}



export default App;
