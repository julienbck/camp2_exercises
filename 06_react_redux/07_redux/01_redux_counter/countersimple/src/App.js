import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';

class App extends Component {
  constructor(props){
    super(props);
    this.state = {
      counter: 0
    }
  }
  increment1 = () => {
    this.setState({
      counter: this.state.counter + 1
    })
  }
  decrement1 = () => {
    this.setState({
      counter: this.state.counter - 1
    })
  }
  increment10 = () => {
    this.setState({
      counter: this.state.counter + 10
    })
  }
  decrement10 = () => {
    this.setState({
      counter: this.state.counter - 10
    })
  }
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <p className="App-intro">
        <button onClick={this.decrement10}> - by 10</button>
          <button onClick={this.decrement1}> - by 1 </button>
          <span>{this.state.counter}</span>
          <button onClick={this.increment1}>+ by 1</button>
          <button onClick={this.increment10}>+ by 10</button>
        </p>
      </div>
    );
  }
}

export default App;
