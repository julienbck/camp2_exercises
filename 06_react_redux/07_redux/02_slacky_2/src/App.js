import React, { Component } from "react";
import "./App.css";
import Login from "./Login";
import Chat from "./Chat";
import { createStore } from 'redux';



class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      userName: null,
      messages: [],
      date: ""
    };
    // Attaching the websocket to our App so we can reuse it
    this.websocket = new WebSocket("ws://localhost:8080");
  }
  componentDidMount() {
    // Listen for messages
    this.websocket.addEventListener("message", event => {
      const message = JSON.parse(event.data);
      console.log("Message from server ", message);
      switch (message.type) {
        case "CONNECTION_START":
        default:
          return;
        case "MESSAGES":
          this.setState({ messages: message.data });
          console.log(message.data);
          return;
      }
    });
  }

  handleUserName = userName => {
    this.setState({ userName: userName });
    this.websocket.send(
      JSON.stringify({
        type: "LOGIN",
        userName: userName
      })
    );
  };

  sendMessage = message => {
    this.websocket.send(
      JSON.stringify({
        type: "NEW_MESSAGE",
        userName: this.state.userName,
        message: message,
      })
    );
  };
  render() {
    return (
      <div className="App">
        <header className="App-header">
            {this.state.userName ? (
            <p>
            Hi {this.state.userName} !</p>
          ) : (
            <p>Welcome
            </p>
          )}
        </header>

        {this.state.userName ? (
          <Chat sendMessage={this.sendMessage} messages={this.state.messages} />
        ) : (
          <Login handleUserName={this.handleUserName} />
        )}
      </div>
    );
  }
}


export default App;
