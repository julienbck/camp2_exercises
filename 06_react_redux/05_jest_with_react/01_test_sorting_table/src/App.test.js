import React from 'react';
import App from './App.js';
import renderer from 'react-test-renderer';
import ReactDOM from "react-dom";
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

const products =
[
  { "decathlon_id": 8282689, "title": "Corne chasse 14cm", "price": 9.99 },
  { "decathlon_id": 8354464, "title": "Basic L print Long Gold Fusion", "price": 9.99 },
  { "decathlon_id": 8380024, "title": "RUN ELIOPRIME", "price": 54.99 },
  { "decathlon_id": 8379970, "title": "Pantalon Gym", "price": 12.99 },
  { "decathlon_id": 8247793, "title": "PALMES WADERS", "price": 24.99 },
  { "decathlon_id": 8357549, "title": "MINIMIZER EDEN UNI  NOIR", "price": 19.99 },
  { "decathlon_id": 8326155, "title": "Pantalon Training mesh marine", "price": 44.99 },
  { "decathlon_id": 8329121, "title": "COUTEAU A PALOURDES", "price": 4.99 },
  { "decathlon_id": 8370749, "title": "Doudoune Hike 100 garçon bleu", "price": 9.99 },
  { "decathlon_id": 8298354, "title": "OREILLER CONFORT", "price": 6.99 },
  { "decathlon_id": 8044622, "title": "2 guêtres RIDING noir", "price": 14.99 },
  { "decathlon_id": 8249674, "title": "BOBINE FUN 2 3 4mm X 40 20 12m", "price": 6.99 },
  { "decathlon_id": 8353265, "title": "Justaucorps manche longue Gym.", "price": 34.99 }
]

beforeAll(() => {
  configure({ adapter: new Adapter() });
})

test("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<App />, div);
});

test("everything looks the same", () => {
  const component = renderer.create(<App />);

  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test("everything looks the same with reverseSort = FALSE", () => {
  const component = renderer.create(<App reverseSort={false} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test("everything looks the same with reverseSort = TRUE", () => {
  const component = renderer.create(<App reverseSort={true} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Sort change when click on ASC Decathlon_id",() => {
  const result = shallow(<App lines={products} />);
  result.find("th").at(0).simulate("click");
  expect(result.find("td").at(0).text()).toBe("8380024");
});
test("Sort change when click to DESC Decathlon_id",() => {
  const result = shallow(<App lines={products} />);
  result.find("th").at(0).simulate("click");
  result.find("th").at(0).simulate("click");
  expect(result.find("td").at(0).text()).toBe("8044622");
});
test("Sort change when click on ASC Title",() => {
  const result = shallow(<App lines={products} />);
  result.find("th").at(1).simulate("click");
  expect(result.find("td").at(1).text()).toBe("2 guêtres RIDING noir");
});
test("Sort change when click to DESC Title",() => {
  const result = shallow(<App lines={products} />);
  result.find("th").at(1).simulate("click");
  result.find("th").at(1).simulate("click");
  expect(result.find("td").at(1).text()).toBe("RUN ELIOPRIME");
});
test("Sort change when click on ASC Price",() => {
  const result = shallow(<App lines={products} />);
  result.find("th").at(2).simulate("click");
  expect(result.find("td").at(2).text()).toBe("4.99");
});
test("Sort change when click to DESC Price",() => {
  const result = shallow(<App lines={products} />);
  result.find("th").at(2).simulate("click");
  result.find("th").at(2).simulate("click");
  expect(result.find("td").at(2).text()).toBe("54.99");
});
