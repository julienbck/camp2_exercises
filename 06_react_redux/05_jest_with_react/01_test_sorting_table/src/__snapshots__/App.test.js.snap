// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`everything looks the same 1`] = `
<table>
  <thead>
    <tr>
      <th
        onClick={[Function]}
      >
        ID
      </th>
      <th
        onClick={[Function]}
      >
        Title
      </th>
      <th
        onClick={[Function]}
      >
        Price
      </th>
    </tr>
  </thead>
</table>
`;

exports[`everything looks the same with reverseSort = FALSE 1`] = `
<table>
  <thead>
    <tr>
      <th
        onClick={[Function]}
      >
        ID
      </th>
      <th
        onClick={[Function]}
      >
        Title
      </th>
      <th
        onClick={[Function]}
      >
        Price
      </th>
    </tr>
  </thead>
</table>
`;

exports[`everything looks the same with reverseSort = TRUE 1`] = `
<table>
  <thead>
    <tr>
      <th
        onClick={[Function]}
      >
        ID
      </th>
      <th
        onClick={[Function]}
      >
        Title
      </th>
      <th
        onClick={[Function]}
      >
        Price
      </th>
    </tr>
  </thead>
</table>
`;
